### ISSUE TITLE
Syntax not compatible in 750: FILTER IN

### ISSUE THREAD

**Issue by jrodriguez-rc** [abaplint/abaplint#1545]
The option `IN" in `FILTER` sentence is not allowed in 750 version:
![image](https://user-images.githubusercontent.com/56556686/98935134-5e1bfa00-24e3-11eb-95b7-e27c67ebafb3.png)

abaplint doesn't detect it.

**Comment by larshp** [comment]
FILTER with IN should work above 740, https://help.sap.com/doc/abapdocu_740_index_htm/7.40/en-US/index.htm

Can you try moving the IN to before the USING KEY?

**Comment by jrodriguez-rc** [comment]
I tried with different syntax:
```
  METHOD filter_in.

    TYPES:
      BEGIN OF ts_type,
        msgty TYPE msgty,
      END OF ts_type.


    DATA:
      filter       TYPE STANDARD TABLE OF bapiret2 WITH EMPTY KEY,
      lt_types     TYPE SORTED TABLE OF ts_type WITH UNIQUE KEY msgty,
*                                                  WITH NON-UNIQUE SORTED KEY type COMPONENTS msgty,
      lt_types2 TYPE SORTED TABLE OF msgty WITH UNIQUE KEY table_line,
      lt_messages  TYPE SORTED TABLE OF bapiret2 WITH UNIQUE KEY type,
      lt_messages2 TYPE STANDARD TABLE OF bapiret2 WITH EMPTY KEY
                                                  WITH NON-UNIQUE SORTED KEY type COMPONENTS type.


    filter =
        FILTER #( lt_messages WHERE type = 'E' ).

    filter =
        FILTER #( lt_messages2 IN lt_types2 USING KEY type WHERE type = table_line ).

    filter =
        FILTER #( lt_messages2 USING KEY type WHERE type = 'E' ).

    filter =
        FILTER #( lt_messages2 IN lt_types WHERE type = msgty ).

    filter =
        FILTER #( lt_messages2 IN lt_types USING KEY type WHERE type = msgty ).

    filter =
        FILTER #( lt_messages2 USING KEY type IN lt_types WHERE type = msgty ).

    filter =
        FILTER #( lt_messages2 USING KEY type IN zcl_adu_messages=>message_error_types WHERE type = table_line ).

  ENDMETHOD.
```

Syntax check in 750
![image](https://user-images.githubusercontent.com/56556686/98939954-8529fa00-24ea-11eb-9b7c-002460161a39.png)

Syntax check in 752
![image](https://user-images.githubusercontent.com/56556686/98939989-9410ac80-24ea-11eb-859b-64388458f770.png)
